src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1451,70 +1455,31 @@ private void CheckMemberNameConflicts(DiagnosticBag diagnostics)
-
-                            // The other set might already contain a conversion which would collide
-                            // *as a method* with the current conversion.
-                            if (!conversionsAsMethods.ContainsKey(conversion))
-                            {
-                                conversionsAsMethods[conversion] = conversion;
-                            }
-                        }
-
-                        // Does this conversion collide *as a method* with any previously-seen
-                        // non-conversion method?
-
-                        SourceMethodSymbol previousMethod;
-                        if (methodsBySignature.TryGetValue(conversion, out previousMethod))
-                        {
-                            ReportMethodSignatureCollision(diagnostics, conversion, previousMethod);
-                        }
-                        // Do not add the conversion to the set of previously-seen methods; that set
-                        // is only non-conversion methods.
-
                     }
-                    else if ((object)method != null)
-                    {
-                        // Does this method collide *as a method* with any previously-seen 
-                        // conversion?
 
-                        SourceMethodSymbol previousConversion;
-                        if (conversionsAsMethods.TryGetValue(method, out previousConversion))
-                        {
-                            ReportMethodSignatureCollision(diagnostics, method, previousConversion);
-                        }
-                        // Do not add the method to the set of previously-seen conversions.
-
-                        // Does this method collide *as a method* with any previously-seen
-                        // non-conversion method?
+                    var method = symbol as SourceMethodSymbol;
+                    if ((object)method != null)
+                    {
+                        OneOrMany<SourceMethodSymbol> methods;
+                        methodsBySignature[method] = methodsBySignature.TryGetValue(method, out methods) ?
+                            methods.Add(method) :
+                            OneOrMany.Create(method);
+                    }
+                }
 
-                        SourceMethodSymbol previousMethod;
-                        if (methodsBySignature.TryGetValue(method, out previousMethod))
-                        {
-                            ReportMethodSignatureCollision(diagnostics, method, previousMethod);
-                        }
-                        else
-                        {
-                            // We haven't seen this method before. Make a note of it in case
-                            // we see a colliding method later.
-                            methodsBySignature[method] = method;
-                        }
+                foreach (var group in methodsBySignature.Values)
```

Comment:
Good catch, thanks. This was left over from an earlier attempt to match up replaced methods. I've reverted the change.